### PR TITLE
VSC docker debugging + GUI from withing container 

### PR DESCRIPTION
**Pull request by alior101**
unfortunately still no proper documentation for the added features since it's still WIP but anyone working with VSC insider version will be able to work this out.  Also, matplotlib-cpp is not yet with 3d scatter plot. I will try to PR the original author to include it upstream.  At the moment, running tmexample will just open a sample 3d surface just to show GUI is working from the container...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 cmake_minimum_required(VERSION 3.0)
 
+
 if(NOT CMAKE_BUILD_TYPE)
-	set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+	set(CMAKE_BUILD_TYPE "Debug" CACHE STRING
```
Comment (marty1885) [inline CMakeLists.txt]: Please set default build to Release as most users expect a Release build as default.

**Comment by alior101** [comment]
changed Debug into Release ..

**Comment by marty1885** [comment]
Oops. I found my other comments aren't visible to you. Sorry for the inconveniences.

Since we have decided that matplotlib-cpp seems not to be an ideal solution. Should we wait for a better one?

**Comment by alior101** [comment]
yes. I'm going for another render engine ogre3d ... it looks much better
but I need more time to demo it ... I'll add another example and not ride
on tmexample this time ...

**Comment by marty1885** [comment]
I guess I'll close this PR for now. Fell free to re-open it if you have other related things to push.

**Pull request closed without merging**